### ISSUE TITLE
Preserve grouping of module items

### ISSUE THREAD

**Issue by copy** [ocaml-ppx/ocamlformat#521]
Consider this cut down human-formatted code:

```ocaml
let cmos_rtc_seconds = 0x00
let cmos_rtc_seconds_alarm = 0x01
let cmos_rtc_minutes = 0x02

let log_other = 0x000001
let log_cpu = 0x000002
let log_fpu = 0x000004

let cr0_pe = 1 lsl 0
let cr0_mp = 1 lsl 1
let cr0_em = 1 lsl 2
```

Depending on the configuration, ocamlformat either drops the newlines or adds newlines between every `let` binding, while I'd like to preserve the grouping. I don't consider this a failure of the formatter, as I don't expect the formatter to understand how to group top-level bindings by their semantics, but I do find the human-formatted version much more readable.

**Comment by jberdine** [comment]
Note that if you use `let`-`and` groups of binding you can get the spacing I think you want:
```
let cmos_rtc_seconds = 0x00
and cmos_rtc_seconds_alarm = 0x01
and cmos_rtc_minutes = 0x02

let log_other = 0x000001
and log_cpu = 0x000002
and log_fpu = 0x000004

let cr0_pe = 1 lsl 0
and cr0_mp = 1 lsl 1
and cr0_em = 1 lsl 2
```

**Comment by copy** [comment]
That's very neat and probably works as a work-around for me. It does feel a little bit like misuse of `and` though, would you recommend doing this in real code?

**Comment by jberdine** [comment]
Opinions will vary I expect, but `and` does semantically group bindings together, and the formatting reflects that to some extent. I wouldn't give anyone a hard time during code review over using `and` here. Adding comments would also spread the groups out, which might be more helpful.

**Comment by gpetiot** [comment]
I agree with Josh here `and` can be used for that, but if you have the time to contribute to ocamlformat, you can have a look at `Ast.break_between` it will be easy to add a new option that groups the bindings if they have the same prefix.

**Comment by gpetiot** [comment]
Closed with #538
